src/V3Delayed.cpp

```diff
@@ -246,8 +262,122 @@ class DelayedVisitor final : public VNVisitor {
+                if (VN_IS(selp->lsbp(), Const)) {
+                    lsb = selp->lsbConst();
+                    msb = selp->msbConst();
+                }
+            }
+            if (writeRef.isNBA) {
+                nbaRefs.emplace_back(writeRef.refp, writeRef.inNonComb, lsb, msb);
+            } else {
+                blkRefs.emplace_back(writeRef.refp, writeRef.inNonComb, lsb, msb);
+            }
+        }
+        // We only run this function on targets of NBAs, so there should be at least one...
+        UASSERT_OBJ(!nbaRefs.empty(), vscp, "Did not record NBA write");
+        // If no blocking upadte, then we are good
+        if (blkRefs.empty()) return false;
+
+        // If the blocking assignment is in non-combinational logic (i.e.:
+        // in logic that has an explicit trigger), then we can safely
+        // implement it (there is no race between clocked logic and post
+        // scheduled logic), so need not error
+        blkRefs.erase(std::remove_if(blkRefs.begin(), blkRefs.end(),
+                                     [](const Ref& ref) { return ref.inNonComb; }),
+                      blkRefs.end());
+
+        // If nothing left, then we need not error
+        if (blkRefs.empty()) return true;
+
+        // If not a packed variable, warn here as we can't prove independence
+        if (!isPacked) {
+            const Ref& blkRef = blkRefs.front();
+            const Ref& nbaRef = nbaRefs.front();
+            vscp->v3warn(
+                BLKANDNBLK,
+                "Unsupported: Blocking and non-blocking assignments to same non-packed variable: "
+                    << vscp->varp()->prettyNameQ() << '\n'
+                    << vscp->warnContextPrimary() << '\n'
+                    << blkRef.refp->warnOther() << "... Location of blocking assignment\n"
+                    << blkRef.refp->warnContextSecondary() << '\n'
+                    << nbaRef.refp->warnOther() << "... Location of nonblocking assignment\n"
+                    << nbaRef.refp->warnContextSecondary());
+            return true;
+        }
+
+        // We need to error if we can't prove the written bits are independent
+
+        // Sort refs by interval
+        const auto lessThanRef = [](const Ref& a, const Ref& b) {
+            if (a.lsb != b.lsb) return a.lsb < b.lsb;
+            return a.msb < b.msb;
+        };
+        std::stable_sort(blkRefs.begin(), blkRefs.end(), lessThanRef);
+        std::stable_sort(nbaRefs.begin(), nbaRefs.end(), lessThanRef);
+        // Iterate both vectors, checking for overlap
+        auto bIt = blkRefs.begin();
+        auto nIt = nbaRefs.begin();
+        while (bIt != blkRefs.end() && nIt != nbaRefs.end()) {
+            if (lessThanRef(*bIt, *nIt)) {
+                if (nIt->lsb <= bIt->msb) break;  // Stop on Overlap
+                ++bIt;
+            } else {
+                if (bIt->lsb <= nIt->msb) break;  // Stop on Overlap
+                ++nIt;
+            }
+        }
```

Comment:
I worry about potential performance here.  Wonder if we should make a general e.g. IntervalTree algorithm as think there's a few places we can make good use of it. I'm ok keeping it until it's a problem, maybe factor this into a subroutine in meantime?

Reply:
The way it's written it's O(n*log(n)) (due to the sort, the `while` is O(n)), where n is the number of write references to a variable. It's only a log(n) factor worse than the whole pass, so I'm not worried here for now. I think an interval tree would be the same complexity, albeit maybe easier to read. Let's add when needed elsewhere too.